rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java

```diff
@@ -198,16 +195,24 @@ protected static boolean isValidPegInTx(
                 return false;
             }
 
+            TransactionInput txInput = tx.getInput(index);
```

Comment:
`index` is declared above as `final int index = i;`. A bit weird, shouldn't we simply use `i`?

Reply:
the `scriptCorrectlySpendsTx` method takes a "final int", thats why the casting

Reply:
oh ok, thanks